File: src/main/java/com/rtomyj/skc/dao/ProductDao.java
package com.rtomyj.skc.dao;

import com.rtomyj.skc.enums.ProductType;
import com.rtomyj.skc.model.product.Product;
import com.rtomyj.skc.model.product.ProductContent;
import com.rtomyj.skc.model.product.Products;

import java.util.List;
import java.util.Map;
import java.util.Set;

public interface ProductDao {
    Product getProductInfo(final String productId, final String locale);

    List<Product> getProductsByLocale(final String locale);

    Set<Product> getProductDetailsForCard(final String cardId);

    Set<ProductContent> getProductContents(final String productId, final String locale);

    Products getAllProductsByType(final ProductType productType, final String locale);

    Map<String, Integer> getProductRarityCount(final String packId);
}

File: src/main/kotlin/com/rtomyj/skc/dao/ProductDao.kt
package com.rtomyj.skc.dao

import com.rtomyj.skc.model.Product
import com.rtomyj.skc.model.ProductContent
import com.rtomyj.skc.model.Products
import com.rtomyj.skc.util.enumeration.ProductType

interface ProductDao {
  fun getProductInfo(productId: String, locale: String): Product
  fun getProductsByLocale(locale: String): List<Product>
  fun getProductDetailsForCard(cardId: String): MutableList<Product>
  fun getProductContents(productId: String, locale: String): Set<ProductContent>
  fun getAllProductsByType(productType: ProductType, locale: String): Products
  fun getProductRarityCount(productId: String): Map<String, Int>
}